### PR TITLE
Wassaf/ 1033 add formatting to ci

### LINKED ISSUE #1033: {RFC} Requirement of Actions to enforce Code Formatting.
### Idea summary

We need a code-formatter

### Further details

I think it is high time we start using flake, autopep8 or black. I am personally inclined to use black but if there are standards which the project adapts, I shall make amends.

### PR DESCRIPTION
**Pull request by wassafshahzad**
Closes #1033

### CODE REVIEW

**Review by AbhijithGanesh — CHANGES_REQUESTED** [review]
Remove .py changes please, please rebranch

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -31,3 +31,5 @@ jobs:
       run: |
         cd project
         pytest -p no:warning
+    - name: flake8 Lint
+      run: flake8 project/ --exclude "**/migrations/*.py **/tests/*.py **/tests.py **/__init__.py" --max-line-length 120
```
Comment (brylie) [inline .github/workflows/main.yml]: Add newline
```suggestion
      run: flake8 project/ --exclude "**/migrations/*.py **/tests/*.py **/tests.py **/__init__.py" --max-line-length 120
      
```

**Review by brylie — CHANGES_REQUESTED** [review]
As @AbhijithGanesh points out, this pull request should only include changes to `.github/workflows/main.yml`

Please modify this branch or start a new one so that only `main.yml` is changed.

**Comment by wassafshahzad** [comment]
I'm just gonna start a new branch

**Pull request closed without merging**